source/module_esolver/esolver_ks.cpp

```diff
@@ -849,6 +849,23 @@ ModuleIO::Output_Potential ESolver_KS<T, Device>::create_Output_Potential(int it
 //! the 16th-20th functions of ESolver_KS
 //! mohan add 2024-05-12
 //------------------------------------------------------------------------------
+template <typename T, typename Device>
+void ESolver_KS<T, Device>::set_xc_first_loop(const UnitCell& ucell)
```

Comment:
I think this function can be defined outside of ESolver, it seems to be a function of 'XC_Functional'. We don't want too many details in ESolver.

Reply:
I've moved it into `XC_Functional`.